src/commands/CmdChart.cpp

```diff
@@ -100,7 +100,7 @@ int renderChart (
   // Load the data.
   auto filtering = IntervalFilterAndGroup ({
     new IntervalFilterAllInRange ({ filter.start, filter.end }),
-    new IntervalFilterAllWithTags (filter.tags())
+    new IntervalFilterAllWithTags (filter.tags(), rules.getBoolean("complexFiltering"))
```

Comment:
I would not make this a top-level configuration (like e.g. `verbose`), but put it into a `commands` subgroup. I propose `commands.filtering` with values `simple` (= default) and `complex` (or maybe better: `boolean`).